libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/RX/NetworkInterface.c

```diff
@@ -252,6 +360,42 @@ static void prvEMACDeferredInterruptHandlerTask( void *pvParameters )
                 Call the standard trace macro to log the occurrence. */
                 iptraceETHERNET_RX_EVENT_LOST();
                 clear_all_ether_rx_discriptors(1);
+                configPRINTF( ( "R_ETHER_Read_ZC2: Cleared descriptors\n" ) );
+            }
+        }
+
+        if( xBytesReceived > 0 )
+        {
+            /* A packet was received. No need to check for the PHY status now,
+            but set a timer to check it later on. */
+            vTaskSetTimeOutState( &xPhyTime );
+            xPhyRemTime = pdMS_TO_TICKS( PHY_LS_HIGH_CHECK_TIME_MS );
+            /* Indicate that the Link Status is high, so that
+            xNetworkInterfaceOutput() can send packets. */
+            if( xPHYLinkStatus == 0 )
+            {
+                xPHYLinkStatus = 1;
+                FreeRTOS_printf( ( "prvEMACHandlerTask: PHY LS assume %d\n", xPHYLinkStatus ) );
+            }
+        }
+        else if( ( xTaskCheckForTimeOut( &xPhyTime, &xPhyRemTime ) != pdFALSE ) || ( FreeRTOS_IsNetworkUp() == pdFALSE ) )
```

Comment:
I don't understand the relationship between FreeRTOS_IsNetworkUp(), xPHYLinkStatus, and xReportedStatus.  Does the R_ETHER_LinkProcess(0) change what FreeRTOS_IsNetworkUp() will return?  Could you please add a comment to this section to give clarity?

Reply:
> Does the R_ETHER_LinkProcess(0) change what FreeRTOS_IsNetworkUp() will return?

Yes indirectly. But once the network is up, it will remain up for ever, even when disconnected from the LAN.
It is possible to bring the network down (`eNetworkDownEvent`), but that is rarely used.

It is not your shortcoming if you get lost here. I will try show it with some pseudo C code:
This is how IP-ask gets the network up:

~~~c
    for( ;; )
    {
        if( xNetworkInterfaceInitialise() == pdPASS )
        {
            /* The peripheral has been initialised, and the Link Status is up. */
            break;
        }
        /* 3 seconds sleep. */
        vTaskDelay( ipINITIALISATION_RETRY_DELAY );
    }
    /* From now on, FreeRTOS_IsNetworkUp() will return true. */
    xNetworkUp = true;
~~~

Within `NetworkInterface.c`, I introduced:
~~~c
BaseType_t xPHYLinkStatus;    /* The Link Status. */
BaseType_t xReportedStatus;   /* The Link Status as reported by the callback. */
~~~

Now here is the PHY polling:

~~~c
    /* Renesas driver: */
    void R_ETHER_LinkProcess (uint32_t channel)
    {
        /* Check the Link Status. */
        /* Call a user function */
    }
~~~

~~~c
    /* In NetworkInterface.c : the PHY call-back function: */
    void prvLinkStatusChange( BaseType_t xStatus )
    {
        if( xReportedStatus != xStatus )
        {
            configPRINTF( ( "prvLinkStatusChange( %d )\n", xStatus ) );
            xReportedStatus = xStatus;
        }
    }
~~~

and in the driver task:

~~~c
    for( ;; )
    {
        ulTaskNotifyTake();

        /* Read RX stream. */
        xBytesReceived = R_ETHER_Read_ZC2( 0, &buffer );

        /* PHY polling. */
        R_ETHER_LinkProcess( 0 );

        if( xPHYLinkStatus != xReportedStatus )
        {
            xPHYLinkStatus = xReportedStatus;
            FreeRTOS_printf( ( "prvEMACHandlerTask: PHY LS now %d\n", xPHYLinkStatus ) );
        }
    }
~~~

PS. I will replace all occurrences of `configPRINTF()` with `FreeRTOS_printf()`.

Reply:
> I will replace all occurrences of configPRINTF() with FreeRTOS_printf().

Just done so in this commit: "Use FreeRTOS_printf for debug logging in stead of configPRINTF"

`FreeRTOS_printf()` will normally become an empty statement in release versions.